src/runtime_src/driver/xclng/xrt/user_aws/shim.cpp

```diff
@@ -162,11 +168,13 @@ namespace awsbwhal {
           std::memset(&orig_info, 0, sizeof(struct fpga_mgmt_image_info));
           fpga_mgmt_describe_local_image(mBoardNumber, &orig_info, 0);
 
-          uint64_t xclbin_id_from_sysfs;
-          if( int retVal = xclGetXclBinIdFromSysfs( xclbin_id_from_sysfs ) != 0 )
+          uuid_t xclbin_uuid_from_sysfs;
+          if( int retVal = xclGetXclBinUuidFromSysfs( xclbin_uuid_from_sysfs ) != 0 )
              return retVal;
 
-          if ( (xclbin_id_from_sysfs == 0) || (axlfbuffer->m_uniqueId != xclbin_id_from_sysfs) || checkAndSkipReload(afi_id, &orig_info) ) {
+          if ( (xclbin_uuid_from_sysfs == 0) ||
```

Comment:
xclbin_uuid_from_sysfs will never be 0. Did you mean uuid_is_null(xclbin_uuid_from_sysfs)?